aicsimageio/tests/writers/test_ome_tiff_writer.py

```diff
@@ -1,80 +1,143 @@
+    Test to check that OmeTiffWriter saves arrays that are reflexive with OmeTiffReader
     """
+    writer = OmeTiffWriter(resources_dir / filename, overwrite_file=True)
+    writer.save(image)
+    writer.close()
 
-    def test_writerShapeComparison(self):
-        self.writer.save(self.image)
+    with OmeTiffReader(resources_dir / filename) as test_output_reader:
+        output = test_output_reader.data
 
-        with OmeTiffReader(self.file) as test_output_reader:
-            output = test_output_reader.data
+    assert output.shape == image.shape[1:]
 
-        self.assertEqual(output.shape, self.image.shape[1:])
 
+def test_loadAssertionError(resources_dir):
     """
     Test to check if save() will only accept 3, 4, 5 dimensions for data
     """
+    image_to_save = np.ones((1, 2, 3, 4, 5, 6))
+    with pytest.raises(Exception):
+        writer = OmeTiffWriter(resources_dir / filename, overwrite_file=True)
+        writer.save(image_to_save)
 
-    def test_loadAssertionError(self):
-        image_to_save = np.ones((1, 2, 3, 4, 5, 6))
-        with self.assertRaises(Exception):
-            self.writer.save(image_to_save)
 
+def test_overwriteFile(resources_dir):
     """
     Test to check if save() can overwrite a file
     """
+    with OmeTiffWriter(resources_dir / filename, overwrite_file=True) as writer:
+        writer.save(image)
 
-    def test_overwriteFile(self):
-        with OmeTiffWriter(self.file, overwrite_file=True) as writer:
-            writer.save(self.image)
 
+def test_dontOverwriteFile(resources_dir):
     """
     Test to check if save() will raise error when user does not want to overwrite a file that exists
     """
+    with pytest.raises(Exception):
+        with OmeTiffWriter(resources_dir / filename) as writer:
+            writer.save(image)
 
-    def test_dontOverwriteFile(self):
-        with self.assertRaises(Exception):
-            with OmeTiffWriter(self.file) as writer:
-                writer.save(self.image)
 
+def test_noopOverwriteFile(resources_dir):
     """
     Test to check if save() silently no-ops when user does not want to overwrite exiting file
     """
-
-    def test_noopOverwriteFile(self):
-        with open(self.file, "w") as f:
-            f.write("test")
-        with OmeTiffWriter(self.file, overwrite_file=False) as writer:
-            writer.save(self.image)
-        with open(self.file, "r") as f:
-            line = f.readline().strip()
-            self.assertEqual("test", line)
-
-    def test_big_tiff(self):
-        x = np.zeros((10, 10))
-        assert OmeTiffWriter._size_of_ndarray(data=x) == 10*10*x.itemsize
+    with open(resources_dir / filename, "w") as f:
+        f.write("test")
+    with OmeTiffWriter(resources_dir / filename, overwrite_file=False) as writer:
+        writer.save(image)
+    with open(resources_dir / filename, "r") as f:
+        line = f.readline().strip()
+        assert "test" == line
+
+
```

Comment:
maybe out of scope but where's the 
```
with open(resource_dir / filename, "w") as fp:
        writer = OmeTiffWriter(fp)
        writer.save(image)
```
It's a very common python paradigm

Reply:
I think the current format (line 64 - 65) of:
```python
with OmeTiffWriter(...) as writer:
   writer.save(data)
```
Is, imo, much preferable. 

That said, the way the `OmeTiffWriter` object is laid out, allows for your method just as well if I recall correctly.